services/galley/src/Galley/Env.hs

```diff
@@ -60,45 +66,62 @@ data Env = Env
     _brig :: Endpoint, -- FUTUREWORK: see _federator
     _cstate :: ClientState,
     _deleteQueue :: Q.Queue DeleteItem,
-    _extEnv :: ExtEnv,
+    _extGetManager :: [Fingerprint Rsa] -> IO Manager,
     _aEnv :: Maybe Aws.Env,
     _mlsKeys :: SignaturePurpose -> MLSKeys,
     _rabbitmqChannel :: Maybe (MVar Q.Channel),
     _convCodeURI :: Either HttpsUrl (Map Text HttpsUrl)
   }
 
--- | Environment specific to the communication with external
--- service providers.
-data ExtEnv = ExtEnv
-  { _extGetManager :: (Manager, [Fingerprint Rsa] -> Ssl.SSL -> IO ())
-  }
-
 makeLenses ''Env
 
-makeLenses ''ExtEnv
-
 -- TODO: somewhat duplicates Brig.App.initExtGetManager
-initExtEnv :: IO ExtEnv
-initExtEnv = do
+initExtEnv :: [Fingerprint Rsa] -> IO Manager
+initExtEnv fingerprints = do
   ctx <- Ssl.context
-  Ssl.contextSetVerificationMode ctx Ssl.VerifyNone
+  Just sha <- getDigestByName "SHA256"
   Ssl.contextAddOption ctx SSL_OP_NO_SSLv2
   Ssl.contextAddOption ctx SSL_OP_NO_SSLv3
   Ssl.contextAddOption ctx SSL_OP_NO_TLSv1
   Ssl.contextSetCiphers ctx rsaCiphers
+  Ssl.contextSetVerificationMode
+    ctx
+    Ssl.VerifyPeer
+      { vpFailIfNoPeerCert = True,
+        vpClientOnce = False,
+        vpCallback =
+          Just
+            ( \_b ctx509store -> do
```

Comment:
would it make sense to give this function an name and use it in both places?  (i'm assuming it's the same as in `Brig.App`?  if not that would be an even better reason to factor out the *shared* code.  to make it transparent where the differences are.)

Reply:
I think this was a wart before, apparently it is quite hard to find a sensible place because these are different packages and the functions differ ever so slightly.

Reply:
move it to ssl-utils?

and as i said: if they have subtle differences, then those differences should be factored out to make them visible.

or maybe at least cross-link the two places and explicitly state that we're too lazy to make this pretty?  :)

Reply:
yeah, if you think `ssl-utils` is suited for this, then perhaps it might be a good idea to move it there if possible.

Reply:
I think this is addressed.